google/cloud/storage/internal/curl_request.cc

```diff
@@ -68,11 +68,11 @@ CurlRequest::~CurlRequest() {
   if (factory_) CurlHandle::ReturnToPool(*factory_, std::move(handle_));
 }
 
-StatusOr<HttpResponse> CurlRequest::MakeRequest(std::string const& payload) && {
+StatusOr<HttpResponse> CurlRequest::MakeRequest(absl::string_view payload) && {
   handle_.SetOption(CURLOPT_UPLOAD, 0L);
   if (!payload.empty()) {
     handle_.SetOption(CURLOPT_POSTFIELDSIZE, payload.length());
-    handle_.SetOption(CURLOPT_POSTFIELDS, payload.c_str());
+    handle_.SetOption(CURLOPT_POSTFIELDS, payload.data());
```

Comment:
`string_view::data()` without `string_view::size()` is a red flag.

Reply:
There was a `.length()` already, changed to `.size()`.

Reply:
Ahh ... sorry I missed that.